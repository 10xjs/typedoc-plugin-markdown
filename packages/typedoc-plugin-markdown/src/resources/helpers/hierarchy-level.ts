import { DeclarationReflection, ReferenceType } from 'typedoc/dist/lib/models';

export function hierachyLevel(this: ReferenceType) {
  const reflection = this.reflection as DeclarationReflection;
  const level = reflection && reflection.extendedTypes ? ` *` : '*';
  return level;
}
