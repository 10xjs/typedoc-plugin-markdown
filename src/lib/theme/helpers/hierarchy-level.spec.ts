import { hierachyLevel } from './hierarchy-level';

describe(`hierachyLevel helper`, () => {
  test(`should compile with extendedType`, () => {
    const data = {
      reflection: {
        extendedTypes: {},
      },
    };
    const result = hierachyLevel.call(data);
    expect(result).toMatchSnapshot();
  });

  test(`should compile without extended types`, () => {
    const data = {
      reflection: {},
    };
    const result = hierachyLevel.call(data);
    expect(result).toMatchSnapshot();
  });
});
