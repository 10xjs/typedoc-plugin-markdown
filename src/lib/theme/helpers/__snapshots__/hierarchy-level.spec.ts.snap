// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hierachyLevel helper should compile with extendedType 1`] = `"!spaces  *"`;

exports[`hierachyLevel helper should compile without extended types 1`] = `"*"`;
